File: NYTViewer-project/app/src/main/java/com/gmail/dp/denzay/nytviewer/di/AppModule.java
package com.gmail.dp.denzay.nytviewer.di;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private Context mContext;

    public AppModule (@NonNull Context aContext) {
        mContext = aContext;
    }

    @Provides
    @Singleton
    Context provideContext() {
        return mContext;
    }

    @Provides
    @Singleton
    SharedPreferences provideSharedPreferences(Context aContext) {
        return aContext.getSharedPreferences(aContext.getApplicationInfo().name + "_settings", Context.MODE_PRIVATE);
    }
}

File: NYTViewerMVVM/app/src/main/java/com/gmail/dp/denzay/nytviewer_mvvm/di/modules/AppModule.kt
package com.gmail.dp.denzay.nytviewer_mvvm.di.modules

import android.app.Application
import android.content.Context
import dagger.Module
import dagger.Provides
import io.realm.Realm
import javax.inject.Singleton

@Module(includes = [ViewModelModule::class, NetworkModule::class, UseCasesModule::class, RepositoriesModule::class])
class AppModule {

    @Singleton
    @Provides
    fun provideContext(app: Application): Context = app

    @Provides
    fun provideRealm(): Realm {
        return Realm.getDefaultInstance()
    }

}